File: DeviceService/src/main/java/fact/it/deviceservice/model/Device.java
package fact.it.deviceservice.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "device")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Device {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String houseId;
    private String name;
    private String brand;
    private String model;
    private String status;
    private boolean active;
    }

File: frontend/src/app/device.ts
export interface Device {
    id: number;
    houseId: string;
    name: string;
    brand: string;
    model: string;
    status: string;
    active: boolean;
}

export interface DevicePost {
    houseId: string;
    name: string;
    brand: string;
    model: string;
    status: string;
    active: boolean;
}